### DISCUSSION TITLE
Adding Support for python 3.12 in wheels

### DISCUSSION THREAD

**Discussion by mworion** [brandon-rhodes/python-sgp4#123] [General]
Hi Brandon,
as python versions moving forward, I looked how to include python 3.12 in the GitHub action. So far, building all variants of wheels getting more and more complicated. My intension would be to drop 32bit support for windows and linux as we don't have it for aarch64 and macOS anyway and update the underlying build process.
What do you think about it ?
Best regards
Michel

**Comment by mworion** [comment]
Done

**Reply by lvzhengcq** [reply]
As of my last knowledge update in September 2021, Python 3.12 had not been released. Therefore, my information might be outdated. However, I can provide you with general guidance on adding support for a new Python version in the context of creating wheels.

Here are the general steps you can follow to add support for Python 3.12 in wheels:

Check Compatibility:
Before making any changes, ensure that the libraries and dependencies your project relies on are compatible with Python 3.12. Check the release notes and documentation of your project's dependencies to see if they officially support Python 3.12.
Update Metadata:
In your project's [setup.py](http://setup.py/) or pyproject.toml, update the metadata to include Python 3.12 in the list of supported Python versions. For example:

python
# setup.py
from setuptools import setup

setup(
    # other metadata
    python_requires='>=3.6, <=3.12',
    # other configuration
)
or for pyproject.toml:

toml
[build-system]
requires = ["setuptools", "wheel"]
build-backend = "setuptools.build_meta"

[project]
# other metadata
python = "^3.6, <=3.12"
# other configuration
Update CI/CD Configuration:
If you have continuous integration (CI) or continuous deployment (CD) processes, update the configuration to include testing against Python 3.12.
Test:
Run your test suite against Python 3.12 to ensure that your project works as expected. Resolve any issues that arise.
Update Documentation:
If your project has documentation, update it to reflect support for Python 3.12.

**Reply by mworion** [reply]
@lvzhengcq, thanks for the hints. Python3.12 support already added.
